### PR TITLE
[inductor][test] Check if mkl dnn bf16 is supported when using bf16

### PR DESCRIPTION
**Pull request by henrylhtsang**
Sometimes the test is run with older cpu, e.g. Intel(R) Xeon(R) CPU E5-2680 v4. If we inspect its `lscpu`, in the flags, we don't see a `avx512_bf16`. So that probably means bf16 is not supported for those hardwares, and hence the unit test can fail. So we add the check in the code.

Context: https://github.com/pytorch/pytorch/pull/135038

Differential Revision: D62984129

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by henrylhtsang** [comment]
@jgong5 @XuehaiPan can I get a stamp?

File: test/inductor/test_cpu_select_algorithm.py [hunk 1]
```diff
@@ -296,7 +278,10 @@ def forward(self, x):
         self.assertEqual(counters["inductor"]["select_algorithm_autotune"], 1)
         if (
             (
-                dtype == torch.bfloat16
+                (
+                    dtype == torch.bfloat16
```
Comment (XuehaiPan) [inline test/inductor/test_cpu_select_algorithm.py:282]: nit: you can merge these two with:

```python
(
    dtype in (torch.bfloat16, torch.float16)
    and torch.ops.mkldnn._is_mkldnn_fp16_supported()
)
```

**Review by XuehaiPan — APPROVED** [review]

File: test/inductor/test_cpu_select_algorithm.py (hunk 1 above)
Comment (henrylhtsang) [reply test/inductor/test_cpu_select_algorithm.py:282]: @XuehaiPan from what my debugging printouts, `torch.ops.mkldnn._is_mkldnn_fp16_supported()` seems always False, while `torch.ops.mkldnn._is_mkldnn_bf16_supported()` can be True or False.

**Review by chenyang78 — APPROVED** [review]

**Pull request closed without merging**